### ISSUE TITLE
AWS CoreOS AMI : missing EBS metadatas

### ISSUE THREAD

**Issue by arthur-c** [coreos/bugs#954]
Hi,

I have nodes with EBS volumes but I'm unable to retrieve all EC2 metadatas, especially EBS informations.

From ami-05783d60 and ami-c36583a8:
```
$ curl http://169.254.169.254/latest/meta-data/block-device-mapping/
ami
ephemeral0
```

If I look on the metadatas from another linux AMI (centos here) with an EBS volume, I get this:

```
curl http://169.254.169.254/latest/meta-data/block-device-mapping/
ami
ebs2
```

http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-metadata.html

**Comment by crawford** [comment]
The EBS volumes only show up in the block device mapping if the AMI has specified these devices (and the CoreOS AMIs do not). I took a snapshot of the latest Alpha image and added an EBS volume onto it. When I boot it, I see:

```
$ curl http://169.254.169.254/latest/meta-data/block-device-mapping/
ami
ebs5
root
```

... where "ebs5" is the device at /dev/xvdf.

As I understand it, the mapping is useful for providing a name to an EBS volume. Since CoreOS doesn't provide these volumes by default, we also don't provide a mapping for potential future volumes. What are you trying to do with this mapping?

**Comment by arthur-c** [comment]
Thanks for taking time on this issue.

I'm using EBS for container permanent storage and I manage the volumes through fleet, Ansible and ec2 modules (ec2_facts, ec2_vol, ec2_snapshots). I want to make a snapshot management timer unit for each service unit with permanent storage. I specifically need the EBS id and device map to be able to perform this operation.

I might be able to find another way to get this infos , but it could be very useful to have this in the AMI CoreOS.

**Comment by crawford** [comment]
OK, that makes sense. Unfortunately, this is going to be a lower priority for us so I encourage you to explore other solutions in the meantime.

**Comment by arthur-c** [comment]
Hi,

I think this ticket can be closed, it's now ok (tested with stable 835.13.0):

```
curl http://169.254.169.254/latest/meta-data/block-device-mapping/    
ami
ebs2
ebs4
```

```
curl http://169.254.169.254/latest/meta-data/                         
ami-id
ami-launch-index
ami-manifest-path
block-device-mapping/
hostname
iam/
instance-action
instance-id
instance-type
local-hostname
local-ipv4
mac
metrics/
network/
placement/
profile
public-hostname
public-ipv4
public-keys/
reservation-id
security-groups
```
